### PR TITLE
use json_build_object instead of string_agg for pg nodes

### PR DESCRIPTION
**Pull request by vjeffrey**
### :nut_and_bolt: Description: What code changed, and why?
the string agg was resulting in mixups where a double quote
inside a tag would cause the entire nodes search api call to fail
bc it couldn't parse the tag.

### :chains: Related Resources
https://github.com/chef/automate/issues/3724

### :+1: Definition of Done
can add a node with a tag that has a double quote in it and still read the nodes list

### :athletic_shoe: How to Build and Test the Change
rebuild nodemanager
navigate to scan jobs/nodes - add node
add a node with a bad tag like `Dot.Comma,Big;\"Trouble`
make sure scan jobs nodes list is still readable

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [n/a] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/nodemanager-service/pgdb/nodes.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ SELECT
   COALESCE(n.source_id, '') AS source_id,
   COALESCE(n.source_region, '') AS source_region,
   COALESCE(n.source_account_id, '') AS source_account_id,
-  COALESCE(('[' || string_agg(DISTINCT '{"key":"' || t.key || '"' || ',"value": "' || t.value || '"}', ',') || ']'), '[]') :: JSON AS tags,
+  COALESCE(json_agg(DISTINCT jsonb_build_object('key', t.key, 'value', t.value)), '[]') AS tags,
```
Comment (vjeffrey) [inline components/nodemanager-service/pgdb/nodes.go:60]: this is the actual change

File: components/nodemanager-service/pgdb/nodes.go [hunk 2]
```diff
@@ -928,7 +928,9 @@ func (db *DB) QueryManualNodesFields(ctx context.Context, filters []*common.Filt
 			results = append(results, node.Name)
 		case "tags":
 			for _, kv := range node.Tags {
-				results = append(results, kv.Key)
+				if len(kv.Key) > 0 {
```
Comment (vjeffrey) [inline components/nodemanager-service/pgdb/nodes.go:931]: i was getting an extra empty string in the test after the pg call change, so i added this to take care of it. open to suggestions on ways to not end up with the empty string in the first place

File: components/compliance-service/api/tests/25_nodes_spec.rb [hunk 3]
```diff
@@ -129,7 +129,8 @@ def cleanup
       ),
       tags: [
         Common::Kv.new(key: "department", value: "marketing"),
-        Common::Kv.new(key: "boss", value: "John")
+        Common::Kv.new(key: "boss", value: "John"),
+        Common::Kv.new(key: "test", value: "Dot.Comma,Big;\"Trouble")
```
Comment (vjeffrey) [inline components/compliance-service/api/tests/25_nodes_spec.rb:133]: this is the troublesome tag added for testing

File: components/compliance-service/api/tests/25_nodes_spec.rb [hunk 4]
```diff
@@ -201,6 +206,7 @@ def cleanup
           "id": node_id1,
           "name": "M$",
           "manager": "automate",
+          "tags": [{}],
```
Comment (vjeffrey) [inline components/compliance-service/api/tests/25_nodes_spec.rb:209]: the switch to json_build_object with json_agg leaves us with an empty array with an empty object inside. this isn't really problematic, but im open to suggestions on how to make it not happen..

File: components/nodemanager-service/pgdb/nodes.go (hunk 2 above)
Comment (lancewf) [reply components/nodemanager-service/pgdb/nodes.go:931]: When you create a new node without any tags a tag with empty key and value is created. The tag can not be deleted. 
image

Comment (lancewf) [reply components/nodemanager-service/pgdb/nodes.go:931]: I found one solution. Add the below to line 256.
```
	if len(tags) == 1 && tags[0].Key == "" && tags[0].Value == "" {
		newNode.Tags = []*common.Kv{}
	} else {
		newNode.Tags = tags
	}
```
But I am still looking to see why the empty object is being created.

Comment (lancewf) [reply components/nodemanager-service/pgdb/nodes.go:931]: For a node with no tags, we are getting back `[{\"key\": null, \"value\": null}]` from the database.

Comment (lancewf) [reply components/nodemanager-service/pgdb/nodes.go:931]: I think I figured it out!
https://stackoverflow.com/questions/24155190/postgresql-left-join-json-agg-ignore-remove-null. 
We use `  COALESCE(json_agg(DISTINCT jsonb_build_object('key', t.key, 'value', t.value)) FILTER (WHERE t.id IS NOT NULL), '[]') AS tags,` to remove the null in the array.

File: components/nodemanager-service/pgdb/nodes.go (hunk 1 above)
Comment (lancewf) [inline components/nodemanager-service/pgdb/nodes.go:60]:
```suggestion
    COALESCE(json_agg(DISTINCT jsonb_build_object('key', t.key, 'value', t.value)) FILTER (WHERE t.id IS NOT NULL), '[]') AS tags,
```

File: components/compliance-service/api/tests/25_nodes_spec.rb (hunk 4 above)
Comment (lancewf) [reply components/compliance-service/api/tests/25_nodes_spec.rb:209]: This should be fixed with the below SQL change.

File: components/nodemanager-service/pgdb/nodes.go (hunk 2 above)
Comment (vjeffrey) [reply components/nodemanager-service/pgdb/nodes.go:931]: nice! i’ll give it a try when i’m done with lunch! thanks!

Comment (vjeffrey) [reply components/nodemanager-service/pgdb/nodes.go]: this worked nicely; thanks! i also updated the key to have a wonky string as discussed in standup

**Pull request merged**